entity_api/src/agreement.rs

```diff
@@ -63,62 +62,15 @@ pub async fn update(db: &DatabaseConnection, id: Id, model: Model) -> Result<Mod
 pub async fn delete_by_id(db: &DatabaseConnection, id: Id) -> Result<(), Error> {
     let result = find_by_id(db, id).await?;
 
-    match result {
-        Some(agreement_model) => {
-            debug!(
-                "Existing Agreement model to be deleted: {:?}",
-                agreement_model
-            );
-
-            agreement_model.delete(db).await?;
-            Ok(())
-        }
-        None => Err(Error {
-            source: None,
-            error_kind: EntityApiErrorKind::RecordNotFound,
-        }),
-    }
-}
-
-pub async fn find_by_id(db: &DatabaseConnection, id: Id) -> Result<Option<Model>, Error> {
-    match Entity::find_by_id(id).one(db).await {
-        Ok(Some(agreement)) => {
-            debug!("Agreement found: {:?}", agreement);
-
-            Ok(Some(agreement))
-        }
-        Ok(None) => {
-            error!("Agreement with id {} not found", id);
-
-            Err(Error {
-                source: None,
-                error_kind: EntityApiErrorKind::RecordNotFound,
-            })
-        }
-        Err(err) => {
-            error!("Error finding Agreement with id {}: {:?}", id, err);
-
-            Err(Error {
-                source: Some(err),
-                error_kind: EntityApiErrorKind::RecordNotFound,
-            })
-        }
-    }
+    result.delete(db).await?;
+    Ok(())
 }
 
-pub async fn find_by(
-    db: &DatabaseConnection,
-    query_filter_map: QueryFilterMap,
-) -> Result<Vec<Model>, Error> {
-    let mut query = Entity::find();
-
-    for column in agreements::Column::iter() {
-        if let Some(value) = query_filter_map.get(&column.to_string()) {
-            query = query.filter(column.eq(value));
-        }
-    }
-
-    Ok(query.all(db).await?)
+pub async fn find_by_id(db: &DatabaseConnection, id: Id) -> Result<Model, Error> {
+    Entity::find_by_id(id).one(db).await?.ok_or_else(|| Error {
+        source: None,
```

Comment:
I love seeing this simplification of the tests that `Entity::find_by_id()` provides. This should make our tests even more reliable in the long run. Nicely done.